tests/acceptance/test_sro_measures_dsl.py

```diff
@@ -0,0 +1,157 @@
+            .filter(practice_registrations.date_end >= index_date)
+        )
+
+        is_registered = filtered_practice_registrations.exists_for_patient()
+
+        # TODO: The following would be nice
+        # not_dead = patient_demographics.date_of_death == None
+
+        # TODO: Does this work as we expect it to work?
+        # not_dead = (
+        #     patient_demographics.select_column(patient_demographics.date_of_death)
+        #     == None  # noqa: E711
+        # )
+
+        # TODO: This does not work, because population.value is an instance of
+        # Comparator
+        # cohort.set_population(is_registered & not_dead)
+
+        cohort.set_population(is_registered)
+
+    with subtest(
+        "practice_id", missing_feature="add practice ids to practice registrations"
+    ):
+        cohort.practice_id = (
+            filtered_practice_registrations.sort_by(practice_registrations.date_end)
+            .last_for_patient()
+            .select_column(practice_registrations.practice_id)
+        )
+        cohort.assert_results(practice_id=1)
+
+    with subtest("codelists"):
+        for measure_name, target_codelist in measures_codelists.items():
+            filtered_clinical_events = (
+                clinical_events.filter(
+                    clinical_events.code.is_in(target_codelist.codes)
+                )
+                .filter(clinical_events.date >= index_date)
+                .filter(clinical_events.date <= end_of_index_month)
+            )
+
+            cohort.add_variable(
+                measure_name, filtered_clinical_events.exists_for_patient()
+            )
+
+            # TODO: SortedEventFrame.exists_for_patient doesn't exist, so we can't
+            # filter and sort together.
+            sorted_clinical_events = filtered_clinical_events.sort_by(
+                clinical_events.date
+            )
+            measure_event_code = f"{measure_name}_event_code"
+            cohort.add_variable(
+                measure_event_code,
+                sorted_clinical_events.last_for_patient().select_column(
+                    clinical_events.code
+                ),
+            )
+
+        cohort.assert_results(
+            medication_review=True,
+            systolic_bp=None,
+            qrisk=None,
+            cholesterol=None,
+            alt=None,
+            tsh=None,
+            rbc=None,
+            hba1c=None,
+            sodium=None,
+            asthma=True,
+            copd=None,
+        )
+
+        cohort.assert_results(
+            medication_review_event_code="394720003",
+            systolic_bp_event_code=None,
+            qrisk_event_code=None,
+            cholesterol_event_code=None,
+            alt_event_code=None,
+            tsh_event_code=None,
+            rbc_event_code=None,
+            hba1c_event_code=None,
+            sodium_event_code=None,
+            asthma_event_code="394720003",
+            copd_event_code=None,
+        )
```

Comment:
These assertions are actually quite expensive (about 1 sec). Could we do these two together?

Reply:
I'd be inclined to pair up the boolean and the code for each codelist.

Reply:
I've paired the boolean and the code, as you suggest.